test/cmdlineTests/standard_debug_info_in_yul_location/output.json

```diff
@@ -172,48 +172,28 @@ object \"C_54\" {
                 {
                     // f()
 
-                    if callvalue() { revert_error_ca66f745a3ce8ff40e2ccaf1ad45db7774001b90d25810abd9040049be7bf4bb() }
-                    abi_decode_tuple_(4, calldatasize())
-                    let ret_0 :=  fun_f_30()
-                    let memPos := allocate_unbounded()
-                    let memEnd := abi_encode_tuple_t_int256__to_t_int256__fromStack(memPos , ret_0)
-                    return(memPos, sub(memEnd, memPos))
+                    external_fun_f_30()
                 }
 
                 case 0x793816ec
                 {
                     // stateVar()
 
-                    if callvalue() { revert_error_ca66f745a3ce8ff40e2ccaf1ad45db7774001b90d25810abd9040049be7bf4bb() }
-                    abi_decode_tuple_(4, calldatasize())
-                    let ret_0 :=  getter_fun_stateVar_10()
-                    let memPos := allocate_unbounded()
-                    let memEnd := abi_encode_tuple_t_int256__to_t_int256__fromStack(memPos , ret_0)
-                    return(memPos, sub(memEnd, memPos))
+                    external_fun_stateVar_10()
                 }
```

Comment:
This probably does not matter given that #10742 but I just realized that this would prevent that optimization (unless this gets inlined back?).

Reply:
Also one of the problems with #10742 was that `switches` with **all** identical cases are apparently rare. But if we removed the `default` here (relying on the fact that cases always end up terminating the execution), it would actually be able to extract the first 2 lines from each case as the common prefix (3 if we moved `memPos` one line above).

Reply:
Yeah, I think moving that part out is just too difficult in general. If we have the more specialized dispatch routine, we could group functions into payable and non-payable and then move the check out or something.

Reply:
Could we just move these two lines outside permanently, i.e. could codegen just put them before the switch?